parsel/selector.py

```diff
@@ -342,6 +357,29 @@ def remove_namespaces(self):
         # remove namespace declarations
         etree.cleanup_namespaces(self.root)
 
+    def remove(self):
+        """
+        Remove matched nodes from the parent element.
+        """
+        try:
+            parent = self.root.getparent()
+        except AttributeError:
+            # 'str' object has no attribute 'getparent'
+            raise CannotRemovePseudoElement(
```

Comment:
Is there a way to check if the element is in fact a pseudo-element? It seems weird to me to raise an exception called `CannotRemovePseudoElement` with something that *might* (also meaning it *might not*) be a pseudo-element.
A similar thing happens with the `CannotRemoveRootElement` exception below.

Reply:
That's a great question, @elacuesta. I'm not really sure if there's another case that would result in elements without a parent or root elements. I've written the messages this way to sound more user-friendly and of course, keep the possibilities open :)

Is there anything I could do to improve this piece of code?

Reply:
The message looks good to me, but maybe it is possible to find a better exception name, to make it clear that it is not necessarily a problem with pseudo elements.

Reply:
Suggestions: 
- `CannotRemoveElementWithNoParent`
- `CannotRemoveElementWithoutParent`

Reply:
Let me know which one you prefer. Or if you have a third option.

Reply:
During the refactoring proccess I found it better to name the exception as `CannotRemoveElementWithoutRoot` because it better reflects what's happening in the source code. Also improved the exception message. 
@elacuesta @kmike

Reply:
Nice, thanks!
Let me exercise my nitpicking skills one last time: what about reusing one of your initial naming suggestions, but for the other exception? i.e. `CannotRemoveRootElement -> CannotRemoveElementWithoutParent`?
It seems to me like that the fact that "the element has no parent" is the one thing you know for certain, while the fact that the node is a `root` element is a possibility (I'm not sure if there are other explanations though).
Thanks again, and sorry for the back and forth.

Reply:
No problem, @elacuesta. That actually makes a lot of sense. I've just pushed another revision. I'm grateful for your time and your valuable thoughts.

Reply:
By all means, thanks to you Victor :tophat: